open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/validate/impl/VRPSubmissionPayloadValidator.java

```diff
@@ -0,0 +1,512 @@
+     * Checks if the given Object is a JSONObject and the JSONObject is non-empty , and it is an instance of a string.
+     *
+     * @param value The Object to be validated.
+     * @return true if the object is a non-null and non-empty JSONObject.
+     */
+    public static boolean isValidString(Object value) {
+        return value instanceof String;
+    }
+
+    /**
+     * Validates initiation parameter in the submission data.
+     *
+     * @param submissionData The JSONObject containing submission data.
+     * @return A JSONObject indicating the validation result.
+     */
+    public static JSONObject validateInitiationParameter(JSONObject submissionData) {
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
+
+        if (submissionData.containsKey(ConsentExtensionConstants.INITIATION)) {
+
+            Object dataObject = submissionData.get(ConsentExtensionConstants.INITIATION);
+
+            if (!isValidJSONObject(dataObject)) {
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_INVALID,
+                        ErrorConstants.INITIATION_NOT_JSON);
+            }
+        } else {
+            log.error(ErrorConstants.INITIATION_NOT_FOUND);
+            return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                    ErrorConstants.INITIATION_NOT_FOUND);
+        }
+
+        return validationResult;
+    }
+
+    /**
+     * Validates instruction parameter in the submission data.
+     *
+     * @param submissionData The JSONObject containing submission data.
+     * @return A JSONObject indicating the validation result.
+     */
+    public static JSONObject validateInstructionParameter(JSONObject submissionData) {
+        JSONObject validationResult = new JSONObject();
+        validationResult.put(ConsentExtensionConstants.IS_VALID_PAYLOAD, true);
+
+        if (submissionData.containsKey(ConsentExtensionConstants.INSTRUCTION)) {
+
+            Object dataObject = submissionData.get(ConsentExtensionConstants.INSTRUCTION);
+            if (!isValidJSONObject(dataObject)) {
+                return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_INVALID,
+                        ErrorConstants.INSTRUCTION_NOT_JSON);
+            }
+        } else {
+            log.error(ErrorConstants.INSTRUCTION_NOT_FOUND);
+            return ConsentValidatorUtil.getValidationResult(ErrorConstants.FIELD_MISSING,
+                    ErrorConstants.INSTRUCTION_NOT_FOUND);
+        }
+
+        return validationResult;
+    }
+
+    /**
+     * Extracts submission data from a JSONObject.
+     *
+     * @param submissionJson The JSONObject containing submission data.
+     * @return A JSONObject indicating the validation result.
+     */
+    public static JSONObject validateSubmissionData(JSONObject submissionJson) {
+        JSONObject validationResult = new JSONObject();
```

Comment:
please refer to https://github.com/wso2/financial-open-banking/pull/58/files#r1632620603 , check other places too

Reply:
Fixed in [a613632]